web/autolinker.js

```diff
@@ -96,31 +96,38 @@ class Autolinker {
   static #regex;
 
   static findLinks(text) {
-    // Regex can be tested and verified at https://regex101.com/r/zgDwPE/1.
+    // Regex can be tested and verified at https://regex101.com/r/rXoLiT/2.
     this.#regex ??=
-      /\b(?:https?:\/\/|mailto:|www\.)(?:[[\S--\[]--\p{P}]|\/|[\p{P}--\[]+[[\S--\[]--\p{P}])+|\b[[\S--@]--\{]+@[\S--.]+\.[[\S--\[]--\p{P}]{2,}/gmv;
+      /\b(?:https?:\/\/|mailto:|www\.)(?:[\S--[\p{P}<>]]|\/|[\S--[\[\]]]+[\S--[\p{P}<>]])+|\b[\S--[@\p{Ps}\p{Pe}<>]]+@([\S--[\p{P}<>]]+(?:\.[\S--[\p{P}<>]]+)+)/gmv;
 
     const [normalizedText, diffs] = normalize(text);
     const matches = normalizedText.matchAll(this.#regex);
     const links = [];
     for (const match of matches) {
-      const raw =
-        match[0].startsWith("www.") ||
-        match[0].startsWith("mailto:") ||
-        match[0].startsWith("http://") ||
-        match[0].startsWith("https://")
-          ? match[0]
-          : `mailto:${match[0]}`;
-      const url = createValidAbsoluteUrl(raw, null, {
+      const [url, emailDomain] = match;
+      let raw;
+      if (
+        url.startsWith("www.") ||
+        url.startsWith("http://") ||
+        url.startsWith("https://")
+      ) {
+        raw = url;
+      } else {
+        if (!URL.canParse(`http://${emailDomain}`)) {
+          continue;
+        }
+        raw = url.startsWith("mailto:") ? url : `mailto:${url}`;
+      }
```

Comment:
Can't this be simplified/improved a little bit, now that the `try-catch` block was removed?
```suggestion
      } else if (URL.canParse(`http://${emailDomain}`)) {
        raw = url.startsWith("mailto:") ? url : `mailto:${url}`;
      } else {
        continue;
      }     
```

Reply:
Fixed, thanks!